Tests/SonarScanner.MSBuild.Common.Test/AggregatePropertiesProviderTests.cs

```diff
@@ -94,5 +96,33 @@ public void AggProperties_Aggregation()
         aggProvider.AssertExpectedPropertyValue("p3.unique.key.1", "p3 unique value 1");
     }
 
+    [TestMethod]
+    public void AggProperties_GetAllPropertiesPerProvider()
+    {
+        var listPropertiesProvider = new ListPropertiesProvider(PropertyProviderKind.SQ_SERVER_SETTINGS);
+        listPropertiesProvider.AddProperty("shared.key.A", "value A from one");
+        listPropertiesProvider.AddProperty("key.B", "value B from one");
+        listPropertiesProvider.AddProperty("p1.unique.key.1", "p1 unique value 1");
+
+        var args = new List<ArgumentInstance>
+        {
+            new(CmdLineArgPropertyProvider.Descriptor, "shared.key.A=value A from one"),
+            new(CmdLineArgPropertyProvider.Descriptor, "p2.unique.key.1=p2 unique value 1")
+        };
+        _ = CmdLineArgPropertyProvider.TryCreateProvider(args, new TestLogger(), out var commandLineProvider);
+
+        var aggProvider = new AggregatePropertiesProvider(commandLineProvider, listPropertiesProvider);
+        var expected = new Dictionary<string, PropertyProviderKind>
+        {
+            { "key.B", PropertyProviderKind.SQ_SERVER_SETTINGS },
+            { "p1.unique.key.1", PropertyProviderKind.SQ_SERVER_SETTINGS },
+            { "shared.key.A", PropertyProviderKind.CLI },
+            { "p2.unique.key.1", PropertyProviderKind.CLI },
+        };
+
+        aggProvider.AssertExpectedPropertyCount(4);
+        aggProvider.GetAllPropertiesWithProvider().ToDictionary(x => x.Key.Id, x => x.Value.ProviderType).Should().BeEquivalentTo(expected);
```

Comment:
https://fluentassertions.com/collections/
```suggestion
        aggProvider.GetAllPropertiesWithProvider().Should().SatisfyRespectively(
            x =>
            {
                x.Key.Id.Should().Be("key.B");
                x.Key.Value.Should().Be("value B from one");
                x.Value.ProviderType.Should().Be(PropertyProviderKind.SQ_SERVER_SETTINGS);
            }, 
            // ...
            //
            // or like so:
            x =>
            {
                x.Should().BeEquivalentTo(new
                {
                    Key = new
                    {
                        Id = "key.B";
                        Value = "value B from one";
                    };
                Value = new { ProviderType = PropertyProviderKind.SQ_SERVER_SETTINGS; };
            }, ...);
        );

```